### DISCUSSION TITLE
use outsite of component

### DISCUSSION THREAD

**Discussion by HasinaNjaratin** [vinissimus/next-translate#814] [Q&A]
Is it possible to use the function outside of component?
I tried `const { t } = useTranslation();` but i get error : 
`React Hook "useTranslation" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.`

**Comment by aralroca** [comment]
`useTranslation` is a React Hook because the lang and namespace are in a React context. You can use [`getT`](https://github.com/vinissimus/next-translate#gett) , is an asynchronous function to load the t function outside components / pages. It works on both server-side and client-side.

Unlike the useTranslation hook, we can use here any namespace, it doesn't have to be a namespace defined in the "pages" configuration. It downloads the namespace indicated as a parameter on runtime.

```js
import getT from 'next-translate/getT'
// ...
export async function getStaticProps({ locale }) {
  const t = await getT(locale, 'common')
  const title = t('title')
  return { props: { title } }
}
```
